scripts/convert_hf_checkpoint.py

```diff
@@ -41,58 +58,85 @@ def permute(w):
             .reshape(dim, dim)
         )
 
-    with torch.no_grad():
-        sd["transformer.wte.weight"].copy_(sd_hf["model.embed_tokens.weight"])
-        sd["transformer.ln_f.scale"].copy_(sd_hf["model.norm.weight"])
-        sd["lm_head.weight"].copy_(sd_hf["lm_head.weight"])
-
-        for i in range(n_blocks):
-            sd[f"transformer.h.{i}.attn.c_proj.weight"].copy_(
-                sd_hf[f"model.layers.{i}.self_attn.o_proj.weight"]
-            )
-
-            sd[f"transformer.h.{i}.attn.c_attn.weight"][:qkv_size] = permute(
-                sd_hf[f"model.layers.{i}.self_attn.q_proj.weight"]
-            )
-            sd[f"transformer.h.{i}.attn.c_attn.weight"][qkv_size:-qkv_size] = permute(
-                sd_hf[f"model.layers.{i}.self_attn.k_proj.weight"]
-            )
-            sd[f"transformer.h.{i}.attn.c_attn.weight"][-qkv_size:] = sd_hf[
-                f"model.layers.{i}.self_attn.v_proj.weight"
-            ]
-
-            sd[f"transformer.h.{i}.mlp.c_fc1.weight"].copy_(
-                sd_hf[f"model.layers.{i}.mlp.gate_proj.weight"]
-            )
-            sd[f"transformer.h.{i}.mlp.c_fc2.weight"].copy_(
-                sd_hf[f"model.layers.{i}.mlp.up_proj.weight"]
-            )
-            sd[f"transformer.h.{i}.mlp.c_proj.weight"].copy_(
-                sd_hf[f"model.layers.{i}.mlp.down_proj.weight"]
-            )
-
-            sd[f"transformer.h.{i}.rms_1.scale"].copy_(
-                sd_hf[f"model.layers.{i}.input_layernorm.weight"]
-            )
-            sd[f"transformer.h.{i}.rms_2.scale"].copy_(
-                sd_hf[f"model.layers.{i}.post_attention_layernorm.weight"]
-            )
+    bin_files = set(el for el in bin_index["weight_map"].values())
+
+    weight_map = {
+        "self_attn.o_proj.weight": "attn.c_proj.weight",
+        "self_attn.q_proj.weight": "attn.c_attn.weight",
+        "self_attn.k_proj.weight": "attn.c_attn.weight",
+        "self_attn.v_proj.weight": "attn.c_attn.weight",
+        "mlp.gate_proj.weight": "mlp.c_fc1.weight",
+        "mlp.up_proj.weight": "mlp.c_fc2.weight",
+        "mlp.down_proj.weight": "mlp.c_proj.weight",
+        "input_layernorm.weight": "rms_1.scale",
+        "post_attention_layernorm.weight": "rms_2.scale",
+        "model.embed_tokens.weight": "transformer.wte.weight",
+        "model.norm.weight": "transformer.ln_f.scale",
+        "lm_head.weight": "lm_head.weight"
+    }
+
+    for bin_file in bin_files:
+        print("Processing", bin_file)
+
+        hf_weights = torch.load(os.path.join(hf_checkpoint_path, bin_file), map_location="cpu")
+
+        with torch.no_grad():
```

Comment:
Convert this context manager to a function decorator so everything is covered. This will also remove it from inside the for loop

Reply:
Good point, done